src/ida_pro_mcp/mcp-plugin.py

```diff
@@ -891,9 +891,15 @@ def set_comment(
     if not idaapi.set_cmt(address, comment, False):
         raise IDAError(f"Failed to set disassembly comment at {hex(address)}")
 
+    if not ida_hexrays.init_hexrays_plugin():
+        return
+
     # Reference: https://cyber.wtf/2019/03/22/using-ida-python-to-analyze-trickbot/
     # Check if the address corresponds to a line
-    cfunc = decompile_checked(address)
+    try:

```

Comment:
This will hide a real decompilation error. Did you observe `ida_hexrays.init_hexrays_plugin()` succeeding but the decompilation failing if you do not have a decompiler available?

Reply:
If you have a HexRays decompiler but not the right decompiler for the executable , the init function still returns true. 

For example, I hit this when reversing an x86 binary but I only have the x64 and ARM64 decompilers.

Reply:
I figured out a way to specifically check for license problems. I introduced a new error type called `DecompilerLicenseError`, which I check in this try now. The other decompilation errors should bubble up.